### DISCUSSION TITLE
Using tangram-es as a library with platform-specific tweaks

### DISCUSSION THREAD

**Discussion by tehKaiN** [tangrams/tangram-es#2283] [Ideas]
In the past, I've done the wxwidgets support by adding platforms/wxwidgets files on a separate branch in my fork. This solution was flawed, because it needed separate set of hacks when building on windows/linux/rpi-linux. So I thought that I could do as follows:

- create wxTangram repo
- tangram-es is a submodule of wxTangram
- when configuring cmake, the tangram-es auto-deduces windows/linux/rpi
- thus configured tangram lib is used by wxTangram as a base

And this is problematic because currently, the tangram-es' CMake exposes following targets:

- tangram-core - base library, without platform-dependent stuff (gl_platform ifdefs, urlclient)
- trangram - executable with platform-specific tweaks (cmake vars, defines which switch things on and off, platform-specific link libraries)

What I would like to propose is as follows:

- add intermediate step of libtangram, which would consist of tangram-core and platform specific tweaks (platform_gl, urlclient), 
- add separate target of glfwApp which would use libtangram

but is this feasible for all the platforms (ios, android)?

**Comment by mathisloge** [comment]
I've needed this aswell and had a local fork of this with some more additions.  I've added a example in #2280

**Comment by matteblair** [comment]
This seems like a useful idea. I'm going to move this into a discussion, I think that's more appropriate than an issue at this point.
